features/ui/button/button.tsx

```diff
@@ -18,3 +18,260 @@ export const Button = styled.button`
+            color: ${color("gray", 700)};
+            box-shadow: ${shadow("xs")}, 0px 0px 0px 4px ${color("gray", 100)};
+          `
+        : props.color == "empty"
+        ? css`
+            color: ${color("primary", 700)};
+          `
+        : props.color == "emptyGray"
+        ? css`
+            color: ${color("gray", 500)};
+          `
+        : props.color == "error"
+        ? css`
+            background: ${color("error", 600)};
+            border: 1px solid ${color("error", 600)};
+            box-shadow: ${shadow("xs")}, 0px 0px 0px 4px ${color("error", 100)};
+          `
+        : css`
+            background: ${color("primary", 700)};
+            border: 1px solid ${color("primary", 700)};
+            box-shadow: ${shadow("xs")},
+              0px 0px 0px 4px ${color("primary", 100)};
+          `
+      : props.state == "disabled"
+      ? props.color == "secondary"
+        ? css`
+            background: ${color("primary", 25)};
+            border: 1px solid ${color("primary", 25)};
+            color: ${color("primary", 300)};
+            box-shadow: ${shadow("xs")};
+          `
+        : props.color == "gray"
+        ? css`
+            background: #fff;
+            border: 1px solid ${color("gray", 200)};
+            color: ${color("gray", 300)};
+            box-shadow: ${shadow("xs")};
+          `
+        : props.color == "empty"
+        ? css`
+            color: ${color("gray", 300)};
+          `
+        : props.color == "emptyGray"
+        ? css`
+            color: ${color("gray", 300)};
+          `
+        : props.color == "error"
+        ? css`
+            background: ${color("error", 200)};
+            border: 1px solid ${color("error", 200)};
+            box-shadow: ${shadow("xs")};
+          `
+        : css`
+            background: ${color("primary", 200)};
+            border: 1px solid ${color("primary", 200)};
+            box-shadow: ${shadow("xs")};
+          `
+      : null}
+
+${(props) => {
+    switch (props.icon) {
+      case ButtonIcon.only:
+        return css`
+          padding: 10px;
+        `;
+    }
+  }}
+`;
+
+export enum ButtonSize {
+  sm = "sm",
+  md = "md",
+  lg = "lg",
+  xl = "xl",
+}
+
+export enum ButtonColor {
+  primary = "primary",
+  secondary = "secondary",
+  gray = "gray",
+  empty = "empty",
+  emptyGray = "emptyGray",
+  error = "error",
+}
+
+export enum ButtonState {
+  default = "default",
+  hover = "hover",
+  focused = "focused",
+  disabled = "disabled",
+}
+
+export enum ButtonIcon {
+  none = "none",
+  leading = "leading",
+  trailing = "trailing",
+  only = "only",
+}
+
+type ButtonProps = {
+  children: React.ReactNode;
+  size?: ButtonSize;
+  color?: ButtonColor;
+  state?: ButtonState;
+  icon?: ButtonIcon;
+};
+
+export function StyledButton({
+  children,
+  size = ButtonSize.md,
+  state = ButtonState.default,
+  color = ButtonColor.primary,
+  icon = ButtonIcon.none,
+}: ButtonProps) {
```

Comment:
This is fine and I wouldn't expect more. Just FYI, this brings a few problems with it. You defined a set of props but UI kit components need to be flexible. For example, a developer may want to use this component but style it with a `className` or `style` prop. Maybe they want to add a `aria-label` or `data-testid`. Or even worse, they may want to use the button as a link. All of this isn't possible at the moment. 

This code creates a flexible button component. No need to understand it fully but very useful in case you build a UI kit at some point 😄 

```
type ButtonPropsOwn<T extends React.ElementType> = {
  as?: T;
};

type ButtonProps<T extends React.ElementType> = ButtonPropsOwn<T> &
  Omit<React.ComponentPropsWithRef<T>, keyof ButtonPropsOwn<T>>;

export const Button = <T extends React.ElementType = "button">({
  children,
  ...props
}: ButtonProps<T>) => {
  return <button {...props}>{children}</button>;
};

<Button as="a" href="https://google.com">
  Button
</Button>;
```